mantidproject/mantid
# Engg:: Write out calibration file twice to avoid race condition

Description of work.
The engineering interface currently runs the calibration and saves out the files to both 
`User/<RB Number>/Calibration` and `Calibration` in the EnginX folder. 
Previously the implementation saved the file in `User` then copied that file to `Calibration` in the top level directory, however as the saving method is a Python script and the copy step was C++ there was no easy way to verify the Python script executed before the C++. This meant that especially on slower machines the file would not exist when we came to copy it so the file would only exist in one location.

To avoid this we now save the file twice in both locations - whilst this incurs some additional overhead the calibration procedure can take some time so a minor time increase will not be noticeable.

**To test:**
- Go to the EnginX folder - this will be located at either `C:\EnginX_Mantid` or `~/EnginX_Mantid` and delete and existing calibrations from the top level `Calibration` folder and `User/<RB Number>/Calibration`
- Copy the files from `\\olympic\Babylon5\Public\David Fairbrother\z_test17296` for testing
- Open the Engineering Diffraction interface
- Open the calibration tab
- Take note of the RB number you have entered at the top of the screen
- Enter the following:
  Vanadium run : `236516`
  Calibration Sample: `241391`
- Run the calibration
- On completion check that 3 `.prm` files are created in the `Calibration` and `User/<RB Number>/Calibration` folders.

Fixes #17296 .

Does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
